frontend/lib/models/test_execution.dart

```diff
@@ -16,6 +17,7 @@ class TestExecution with _$TestExecution {
     @JsonKey(name: 'c3_link') required String? c3Link,
     required TestExecutionStatus status,
     required Environment environment,
+    @JsonKey(name: 'test_results') required List<TestResult>? testResults,
```

Comment:
if we drop optional `None` type on backend then we don't need to handle nullable here